### PR TITLE
CAL-338 Improve input validation in TreUtility

### PR DESCRIPTION
**Pull request by roelens8**
#### What does this PR do?
-Add input validation in "convertToFloat" and "convertToInt" using a regular expressions.
#### Who is reviewing it? 
@harrison-tarr 
@derekwilhelm 
#### Choose 2 committers to review/merge the PR.
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl
@jaymcnallie
#### How should this be tested?
Run a full build, make sure itests pass. Ingest a NITF containing bad data, and verify that no exceptions occur.
#### What are the relevant tickets?
[CAL-338](https://codice.atlassian.net/browse/CAL-338)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java [hunk 1]
```diff
@@ -38,6 +39,10 @@
     private static final FastDateFormat DATE_FORMATTER = FastDateFormat.getInstance(TRE_DATE_FORMAT,
             TimeZone.getTimeZone("GMT"));
 
+    private static Pattern FLOAT_PATTERN = Pattern.compile("([+-]?\\d*\\.\\d+)(?![-+0-9\\.])");
```
Comment (derekwilhelm) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java]: This regex requires a . to be present in the string.  A float does not require a decimal value to be considered a float.  This regex will cause the convertToFloat method to return null if the evaluated Tre field is "3" when it should return 3.0.  Additionally, this regex doesn't consider exponent notation for floating point numbers.  Therefore, a value of "3e-3" would return null as well even though the correct and valid float value is 0.003.  Not sure if these are really options in a TRE field but a better regex to match a float would be: [-+]?\\d*\\.?\\d+([eE][-+]?\\d+)?

Comment (roelens8) [reply catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java]: Thanks, I'll use your regex instead.

Comment (roelens8) [reply catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java]: My regex for "convertToFloat" has been updated.

**Comment by harrison-tarr** [comment]
Could you add some tests testing expected good and bad inputs?

**Comment by harrison-tarr** [comment]
Also it looks like Floats can technically have underscores in them?? https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html

**Comment by roelens8** [comment]
@harrison-tarr I'll add some tests and give the regex from the link you provided a shot.

**Comment by derekwilhelm** [comment]
I'm pretty sure that underscores are only valid in code.  Having an underscore in the string value that is being parsed would not be allowed.

**Comment by derekwilhelm** [comment]
Performed a full build.  All unit tests and itests passed successfully.

**Comment by jaymcnallie** [comment]
Built branch locally and ingested two NITFS that were identical with the exception of a field that had invalid float values.  Verified the correct NITF parsed and added the attribute to the metacard while the incorrect NITF parsed all but the field with incorrect float values (alpha characters in the field) and did not add the attribute to the metacard but did continue parsing and was discoverable in the catalog.

**Pull request merged**